presto-client/src/main/java/com/facebook/presto/client/FixJsonDataUtils.java

```diff
@@ -106,13 +110,36 @@ private static Object fixValue(TypeSignature signature, Object value)
             TypeSignature keySignature = signature.getTypeParametersAsTypeSignatures().get(0);
             TypeSignature valueSignature = signature.getTypeParametersAsTypeSignatures().get(1);
             Map<Object, Object> fixedValue = new HashMap<>();
+            if (value.getClass() == String.class) {
+                int bracketCount = 0;
+                String parseValue = value.toString();
+                Map<Object, Object> newObj = new HashMap<>();
+                for (int i = 0; i < parseValue.length(); i++) {
+                    if (parseValue.charAt(i) == '{') {
+                        bracketCount += 1;
+                    }
+                    else if (parseValue.charAt(i) == '}') {
+                        bracketCount -= 1;
+                    }
+                    else if (parseValue.charAt(i) == '=' && bracketCount == 1) {
+                        String k = parseValue.substring(1, i);
+                        String v = parseValue.substring(i + 1, parseValue.length() - 1);
+                        newObj.put(k, v);
+                        break;
+                    }
+                }
+                value = newObj;
+            }
```

Comment:
We definitely don't want to manually parse JSON.  Can you try to use that `jsonCodec` and first parse to see if this returns a map, then attempt to recursively deserialize it?